### DISCUSSION TITLE
One to Many relationship CRUD operation

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#9217] [Q&A] [answered]
Hi everyone ,

I am struggling to create a record with specific data. 
Let me explain, I have two models : 

```prisma
model Retrait {
  id        String   @id @default(cuid())
  remboursementId Int?
  remboursement Remboursement? @relation(fields: [remboursementId], references: [id])
}

model Remboursement{
  id        Int    @id @default(autoincrement())
  retraits Retrait[]
}
```

When I want to create a new "Remboursement" directly including some Retrait it does not work. 

Example , ""cktbibix8163814ll80is94iv" is a valid Id and the following code throws erros :  
```ts
 const remboursement = await prisma.remboursement.create({
        data: {
          retraits:"cktbibix8163814ll80is94iv",
        },
      });
```

Error : 
```
Argument retraits: Got invalid value 'cktbibix8163814ll80is94iv' on prisma.createOneRemboursement. Provided String, expected RetraitUncheckedCreateNestedManyWithoutRemboursementInput:
type RetraitUncheckedCreateNestedManyWithoutRemboursementInput {
  create?: RetraitCreateWithoutRemboursementInput | List<RetraitCreateWithoutRemboursementInput> | RetraitUncheckedCreateWithoutRemboursementInput | List<RetraitUncheckedCreateWithoutRemboursementInput>
  connectOrCreate?: RetraitCreateOrConnectWithoutRemboursementInput | List<RetraitCreateOrConnectWithoutRemboursementInput>
  createMany?: RetraitCreateManyRemboursementInputEnvelope
  connect?: RetraitWhereUniqueInput | List<RetraitWhereUniqueInput>
}

```

I was thinking to an alternative solution which would be to create the remboursement without any retrait, than update the retrait with the remborusementId just created but that can be quite heavy as one remboursement could be linked to 25 retraits...

Thank you so much in advance for your help :)

**Comment by ryands17** [comment, marked as answer]
You need to add `connect` as follows:

```ts
 const remboursement = await prisma.remboursement.create({
        data: {
          retraits: { connect: { id: "cktbibix8163814ll80is94iv" } }
        },
});
```

Have a look at this page in the docs on how to work with relations: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries

**Reply by laurent512** [reply]
great thank you !!

**Comment by laurent512** [comment]
Hello @ryands17 

Is there a way to update the connected "retraits" directly in a single query? 
e.g. once the  retraits with  ID "cktbibix8163814ll80is94iv"  is connected, put its "status" to "COMPLETED" ?

**Reply by laurent512** [reply]
Something like 

```
 const remboursement = await prisma.remboursement.create({
        data: {
          retraits: { connect: { id: "cktbibix8163814ll80is94iv" ,
                                          update : { status : "COMPLETED"}
                                        }
                        }
        },
});
```

**Reply by laurent512** [reply]
and the complexity comes as well because many retraits ID will be inserted at once, so all of those will have to see their status updated to "COMPLETED"

**Reply by laurent512** [reply]
and is there a way as well to throw an error if the retraits are already connected ? (so it does not rewrite it ? )

**Reply by ryands17** [reply]
You would need to do this in 2 queries:

1. Set all the relations using `connect`.
2. Update all the relations using `updateMany`.

You can perform this inside a `prisma.$transaction` block so that all the changes are executed.

> and is there a way as well to throw an error if the retraits are already connected ? (so it does not rewrite it ? )

Connecting again doesn't throw an error so you won't be facing this case.

**Reply by laurent512** [reply]
thank you so much, I don't know that much about the $transaction block, but will read it !

But concerning the error, I would like to avoid that "reconnection" works. 
For me , if a query ask for connecting something that is already connected, it means that a mistake occured somewhere. 
Is there a way to do that?

**Reply by ryands17** [reply]
That's fine. If it's occurring again on the same relation, it won't make any changes. I don't think there's a way to detect that without checking if the item exists.

**Reply by laurent512** [reply]
I see thank you.

Concerning the "Update all the relations using updateMany." 
I thought about

>  prisma.withdrawal.updateMany({
>         where: {
>           remboursementId : **X**
>         },
>         data: {
>           status: "COMPLETED",
>         },
>       });

But this will not work as I can not anticipate the ID which will be given to the newly created Remboursement right ?

**Reply by ryands17** [reply]
You're not creating the reimbursement but connecting it, so you already have the `id` in that case.

**Reply by laurent512** [reply]
Hello ryands17, thanks for you help, I think I will repost a new and more clear question because now I am lost ^^'

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
